zerver/lib/upload.py

```diff
@@ -205,9 +236,25 @@ def upload_avatar_image_local(user_file, user_profile, email):
 if settings.LOCAL_UPLOADS_DIR is not None:
     upload_message_image = upload_message_image_local
     upload_avatar_image  = upload_avatar_image_local
+    delete_message_image = delete_message_image_local
+
 else:
     upload_message_image = upload_message_image_s3
     upload_avatar_image  = upload_avatar_image_s3
+    delete_message_image = delete_message_image_s3
+
+
+def claim_attachment(path_id, message):
+    #update attachment entry in DB
+    try:
+        attachment = Attachment.objects.get(path_id=path_id)
+        attachment.message = message
+        attachment.save()
+        return True
```

Comment:
I think we should add a comment here of the form `TODO: Improve logic for the case that the message is already claimed`